sensor/kubernetes/upgrade/process.go

```diff
@@ -400,7 +400,7 @@ func (p *process) checkPodStatus(pod *v1.Pod) *central.UpgradeCheckInFromSensorR
 	} else if waitingState := upgraderContainerStatus.State.Waiting; waitingState != nil {
 		if isImagePullRelatedReason(waitingState.Reason) {
 			s.Error = &central.UpgradeCheckInFromSensorRequest_PodErrorCondition{
-				Message:      fmt.Sprintf("Error pulling image: %s (%s)", waitingState.Reason, waitingState.Message),
+				Message:      fmt.Sprintf("The upgrader had trouble pulling the new \"%s\" image. This typically happens when the image does not exist or Sensor lacks required credentials to pull it.", upgraderContainerStatus.Image),
 				ImageRelated: true,
```

Comment:
Thank you for adding remediation! I'd suggest to add it only to the reason we know it applies to, `ErrImagePull`, and keep the generic message for the rest. Something like 

```
var string remediation
switch waitingState.Reason {
    case "ErrImagePull", "ImagePullBackOff", "ErrImageNeverPull": remediation = " This typically happens when the image does not exist or Sensor lacks required credentials to pull it."
}

<...>
Message: fmt.Sprintf("The upgrader had trouble pulling the new \"%s\" image.%s (Reason: %s)", upgraderContainerStatus.Image, remediation, waitingState.Reason)
```

Reply:
Done. I added the function `getImageErrorRemediation` so that in the future is cleaner to extend if we want to add more remediations.

Reply:
I did a little bit of digging in the differences between `ImagePullBackOff`, `ErrImagePull`, and `ErrImageNeverPull`:

- `ErrImagePull`: Is an error pulling the image.
- `ImagePullBackOff`: Is the state of waiting grace period after an `ErrImagePull`.
- `ErrImageNeverPull`: For what I could find this happens if the image was never pulled `imagePullPolicy: Never` (and it's not present in the cluster?).

With this I would definitely add `ErrImagePull` but I think I'll leave `ErrImageNeverPull` out. WDYT?

Reply:
Sounds good, thank you. We can always extend the switch block as we learn about new edge cases.